app/App.hs

```diff
@@ -15,39 +12,46 @@ import PSR.ContextBuilder
 import PSR.Streaming
 import Streamly.Data.Fold.Prelude qualified as Fold
 import Streamly.Data.Stream.Prelude qualified as Stream
-import System.Exit (exitFailure)
+import Text.Pretty.Simple
 
 --------------------------------------------------------------------------------
 -- Main
 --------------------------------------------------------------------------------
 
+compactPrintOpts :: OutputOptions
+compactPrintOpts =
+    defaultOutputOptionsDarkBg
+        { outputOptionsCompact = True
+        , outputOptionsCompactParens = True
+        , outputOptionsIndentAmount = 2
+        , outputOptionsStringStyle = Literal
+        }
+
 main :: IO ()
 main = do
     Options{..} <- execParser psrOpts
-    CM.ConfigMap{..} <- decodeFileThrow scriptYaml
-    -- TODO: Remove
-    -- TODO: Is there really no better way to parse a hash???
-    hash <- case C.deserialiseFromJSON "\"082efb838a6a38c435bf5a1c823569ad224fab42034d265b95f88665ee27877f\"" of
-        Left err -> do
-            print err
-            exitFailure
-        Right h -> pure h
-    let points = maybe [C.ChainPoint 173166134 hash] pure start -- [ChainPointAtGenesis]
+    config@CM.ConfigMap{..} <- either error pure =<< CM.readConfigMap scriptYaml
+
+    -- TODO: What's a better default here?
+    let points = maybe [C.ChainPointAtGenesis] pure cmStart
     -- TODO: Use a logging interface instead of using putStrLn.
     putStrLn "Started..."
 
-    let confPolicyMap = Map.fromList [(CM.script_hash x, x) | x <- scripts]
-        conn = mkLocalNodeConnectInfo networkId socketPath
+    let conn = mkLocalNodeConnectInfo networkId socketPath
     streamChainSyncEvents conn points
         & Stream.filter (not . isByron)
         & fmap getEventTransactions
         & Stream.postscanl trackPreviousChainPoint
         -- TODO: Try to replace "concatMap" with "unfoldEach".
         & Stream.concatMap (Stream.fromList . (\(a, b) -> (a,) <$> b))
         & Stream.mapM (mkContext1 conn . uncurry mkContext0)
-        & Stream.filter
-            ( \ctx1@Context1{..} ->
-                not . Map.null . Map.restrictKeys confPolicyMap $
-                    Set.union (getMintPolicies context0) (getSpendPolicies ctx1)
+        & Stream.mapMaybe (mkContext2 config)
+        & Stream.tap
```

Comment:
Is there a reason you `tap` the stream here?
Ah, I think you wanted to `trace` instead.